src/zope/i18n/gettextmessagecatalog.py

```diff
@@ -22,7 +23,24 @@
 class _KeyErrorRaisingFallback(object):
     def ugettext(self, message):
         raise KeyError(message)
+
+    def ungettext(self, singular, plural, n):
+        raise KeyError(singular)
+
     gettext = ugettext
+    ngettext = ungettext
+
+
+def plural_formatting(func):
+    @wraps(func)
+    def pformat(catalog, singular, plural, n, *args, **kwargs):
+        msg = func(catalog, singular, plural, n, *args, **kwargs)
+        try:
+            return msg % n
+        except TypeError:
+            # The message cannot be formatted : return it "raw".
+            return msg
```

Comment:
Perhaps expand that this is useful when you want to provide a translations like "One apple" vs "%d apples", where one of the plural forms lacks the %d?  So people who see this later don't have to wonder about the why?

BTW did I guess the reason correctly?